plugins/destination/postgresql/client/transformer.go

```diff
@@ -42,24 +45,28 @@ func (*Client) TransformInt8Array(v *schema.Int8Array) interface{} {
 	for _, d := range v.Dimensions {
 		r.Dimensions = append(r.Dimensions, pgtype.ArrayDimension{Length: d.Length, LowerBound: d.LowerBound})
 	}
-	return &r
+	return &r, nil
 }
 
-func (*Client) TransformJSON(v *schema.JSON) interface{} {
+func (*Client) TransformJSON(v *schema.JSON) (interface{}, error) {
 	return &pgtype.JSON{
 		Bytes:  v.Bytes,
 		Status: cqStatusToPgStatus[v.Status],
-	}
+	}, nil
 }
 
-func (*Client) TransformText(v *schema.Text) interface{} {
+func (*Client) TransformText(v *schema.Text) (interface{}, error) {
+	if strings.Contains(v.Str, "\x00") {
+		return nil, fmt.Errorf("postgres TEXT field cannot contain null bytes")
```

Comment:
I don't think `TransformText` should return errors. just replace it with space or remove. That's the point of transformation. Lets' say we have a database were it doesn't support arrays inside jsons or whatever else then it should just transform and/or skip it to the best of it's ability. potentially you can log this as well via `c.log` but def no error